### PR TITLE
fix(api): only query for potentially valid tokens

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
If the token id is not an objectID there cannot be a corresponding token.

Ref: https://freecodecamp.sentry.io/issues/6093481611/?project=4508182869508096&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&sort=date&statsPeriod=14d&stream_index=0

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: api/src/exam-environment/routes/exam-environment.ts [hunk 1]
```diff
@@ -102,16 +104,17 @@ async function tokenMetaHandler(
     );
   }
 
-  const payload = jwt.decode(encodedToken) as JwtPayload;
-
-  const examEnvironmentAuthorizationToken =
-    payload.examEnvironmentAuthorizationToken;
+  const tokenId = isObjectID(payload.examEnvironmentAuthorizationToken)
+    ? payload.examEnvironmentAuthorizationToken
+    : null;
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/routes/exam-environment.ts:109]:
```suggestion
  if (!isObjectID(payload.examEnvironmentAuthorizationToken)) {
    void reply.code(418);
    return reply.send(
      ERRORS.FCC_EINVAL_EXAM_ENVIRONMENT_AUTHORIZATION_TOKEN(JSON.stringify(e))
    );
  }
```

File: api/src/exam-environment/routes/exam-environment.ts [hunk 2]
```diff
@@ -102,16 +104,17 @@ async function tokenMetaHandler(
...
-      id: examEnvironmentAuthorizationToken
-    }
-  });
+  const token = tokenId
+    ? await this.prisma.examEnvironmentAuthorizationToken.findUnique({
+        where: {
+          id: tokenId
+        }
+      })
+    : null;
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/routes/exam-environment.ts:117]:
```suggestion
  const token = await this.prisma.examEnvironmentAuthorizationToken.findUnique({
    where: {
      id: payload.examEnvironmentAuthorizationToken
    }
  });
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I prefer this:

**Review by ShaunSHamilton — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]

**Pull request merged**